src/main/scala/cognite/spark/v1/AssetHierarchyBuilder.scala

```diff
@@ -79,23 +79,40 @@ class AssetHierarchyBuilder(config: RelationConfig)(val sqlContext: SQLContext)
   import CdpConnector.ioRuntime
 
   def delete(data: DataFrame): Unit =
-    data.foreachPartition((rows: Iterator[Row]) => {
-      val deletes = rows.map(r => fromRow[DeleteItemByCogniteId](r))
-      Stream
-        .fromIterator[IO](deletes, chunkSize = batchSize)
-        .chunks
-        .parEvalMapUnordered(config.parallelismPerPartition) { chunk =>
-          client.assets
-            .deleteRecursive(
-              chunk.toVector.map(_.toCogniteId),
-              recursive = true,
-              ignoreUnknownIds = true)
-            .flatTap(_ => incMetrics(itemsDeleted, chunk.size))
-        }
-        .compile
-        .drain
-        .unsafeRunSync()
-    })
+    if (config.enableSinglePartitionDeleteAssetHierarchy) {
+      data
+        .repartition(numPartitions = 1)
+        .foreachPartition((rows: Iterator[Row]) => {
+          batchedOperation[DeleteItemByCogniteId, Unit](
```

Comment:
A little more deviation from the other branch than just `repartition`, but on the surface looks ~same with `batchedOperation` helper. If you want to reduce things changed at a time may be worth keeping same code as the other branch (or even wrapping just the `data.` vs `data.repartition(1)` into a conditional)